apps/frontend/src/components/ViewFormsList/ViewFormsList.tsx

```diff
@@ -77,12 +90,23 @@ export default function ViewFormsList() {
             }
           }
           return false;
-        })
-      );
-    } else {
-      setForms(allForms);
+        });
     }
-  }, [searchTerm]);
+
+    // filter by hospital
```

Comment:
Lets split this into separate useEffects. So one that handles search term, one for hospital filtering, and one for state filtering.